monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,398 @@
+
+	return s.puHandler.HandlePUEvent(eventInfo.PUID, monitor.EventCreate)
+}
+
+// Start handles start events
+func (s *UIDProcessor) Start(eventInfo *rpcmonitor.EventInfo) error {
+	s.Lock()
+	defer s.Unlock()
+	contextID := eventInfo.PUID
+	pids, err := s.puToPidEntry.Get(contextID)
+	var runtimeInfo *policy.PURuntime
+	if err != nil {
+		runtimeInfo, err = s.metadataExtractor(eventInfo)
+		if err != nil {
+			return err
+		}
+		publishedContextID := contextID + runtimeInfo.Options().CgroupMark
+		// Setup the run time
+		if err = s.puHandler.SetPURuntime(publishedContextID, runtimeInfo); err != nil {
+			return err
+		}
+
+		defaultIP, _ := runtimeInfo.DefaultIPAddress()
+
+		zap.L().Error("Starting ", zap.String("contextID", contextID), zap.String("Publishing", publishedContextID))
+		if perr := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStart); perr != nil {
+			zap.L().Error("Failed to activate process", zap.Error(perr))
+			return perr
+		}
+
+		err = s.processLinuxServiceStart(eventInfo, runtimeInfo)
+
+		if err != nil {
+			zap.L().Error("ProcessLInuxServiceStart", zap.Error(err))
+			return err
+		}
+
+		s.collector.CollectContainerEvent(&collector.ContainerRecord{
+			ContextID: contextID,
+			IPAddress: defaultIP,
+			Tags:      runtimeInfo.Tags(),
+			Event:     collector.ContainerStart,
+		})
+		entry := &putoPidEntry{
+			Info:               runtimeInfo,
+			publishedContextID: publishedContextID,
+		}
+		entry.pidlist = make(map[string]bool, 20)
+		entry.pidlist[eventInfo.PID] = true
+		s.puToPidEntry.Add(contextID, entry)
+		s.pidToPU.Add(eventInfo.PID, contextID)
+		// Store the state in the context store for future access
+		zap.L().Error("ContextID", zap.String("eventInfo.PID", eventInfo.PID), zap.String("eventInfo.OUID", contextID))
+		return s.contextStore.StoreContext(contextID, &StoredContext{
+			EventInfo: eventInfo,
+			MarkVal:   runtimeInfo.Options().CgroupMark,
+		})
+
+	}
+	zap.L().Error("Adding to existing session", zap.String("contextID", contextID))
+	pids.(*putoPidEntry).pidlist[eventInfo.PID] = true
+	s.pidToPU.Add(eventInfo.PID, eventInfo.PUID)
+	err = s.processLinuxServiceStart(eventInfo, pids.(*putoPidEntry).Info)
+	return err
+
+}
+
+// Stop handles a stop event
+func (s *UIDProcessor) Stop(eventInfo *rpcmonitor.EventInfo) error {
+
+	contextID, err := s.generateContextID(eventInfo)
+	if err != nil {
+		return err
+	}
+	if contextID == "/trireme" {
+		return nil
+	}
+	strtokens := strings.Split(contextID, "/")
+	contextID = "/" + strtokens[len(strtokens)-1]
+	zap.L().Error("UID STOP EventInfo", zap.String("EventInfo.PUID", eventInfo.PUID),
+		zap.String("EventInfo.PID", eventInfo.PID),
+		zap.String("EventType", string(eventInfo.EventType)),
+		zap.Bool("HostService", eventInfo.HostService),
+		zap.String("CGROUP", eventInfo.Cgroup),
+		zap.String("contextID", contextID),
+	)
+	s.Lock()
+	defer s.Unlock()
+	stoppedpid := strtokens[len(strtokens)-1]
```

Comment:
same

Reply:
also you are doing this so many times. I think it’s about time to make a function